scripts/core/helpers/waitForMediaToBeReady.ts

```diff
@@ -19,11 +27,12 @@ export const waitForMediaToLoad = (elements: Array<any>, eventName: string): Pro
         }
     };
 
-    elements.forEach((element) => {
-        // check for error in image src
+    filteredElelemnts.forEach((element: Element) => {
         if (element.tagName === 'IMG') {
+            // check for error in image src
             element.addEventListener('error', eventHandler, {once: true});
         } else {
+            eventName = 'loadedmetadata';
```

Comment:
instead of else, use `else if(element.tagName === 'IMG' || element.tagName === 'VIDEO') `. It's faster to understand exactly which types of elements the else branch is applied to.